src/startServer.js

```diff
@@ -0,0 +1,78 @@
+import cluster from 'cluster';
+import os from 'os';
+
+import {
+  CLUSTER_PROCESS_START,
+  CLUSTER_PROCESS_EXIT,
+  SERVER_LISTEN_ERROR,
+} from '#lib/logger.const';
+
+// dotenv should be called on entry to the application to ensure all `process.env.*` variables are correctly set from '.env'
+const dotenv = require('dotenv');
+
+const DOT_ENV_CONFIG = dotenv.config();
+if (DOT_ENV_CONFIG.error) {
+  throw DOT_ENV_CONFIG.error;
+}
+
+// now `process.env.*` variables are set run the rest of the app
+const http = require('http');
+const nodeLogger = require('#lib/logger.node');
+const app = require('./server').default;
+
+const logger = nodeLogger(__filename);
+const port = process.env.PORT || 7080;
+
+const startApplicationInstance = () => {
+  const server = http.createServer(app);
+  let currentApp = app;
+  server.listen(port, error => {
+    if (error) {
+      logger.error(SERVER_LISTEN_ERROR, {
+        error,
+      });
+    }
+  });
+
+  if (module.hot) {
+    logger.info('✅  Server-side Hot Module Replacement enabled');
+
+    module.hot.accept('./server', () => {
+      logger.info('🔁  Hot Module Replacement reloading `./server`...');
+      server.removeListener('request', currentApp);
+      const newApp = require('./server').default; // eslint-disable-line global-require
+      server.on('request', newApp);
+      currentApp = newApp;
+    });
+  }
+};
+
+const processOnline = worker =>
+  logger.info(CLUSTER_PROCESS_START, `Worker ${worker.id} started`);
+
+const processExit = (worker, code, signal) => {
+  const exitReason = code ? ` with code ${code}` : ` due to signal ${signal}`;
+
+  if (code !== 0 && !worker.exitedAfterDisconnect) {
+    cluster.fork();
```

Comment:
What is `cluster.fork()` doing here? is starting a new process if a node terminates?

Reply:
yep, if it terminates for an unexpected reason we start another process up

Reply:
I think in the case where a worker terminates, it could be a bit safer to recycle the entire instance, it is likely the process has failed due to an instance wide issue e.g. memory exhausted

That would happen naturally if we did not fork new processes when a cluster worker died.

Reply:
Happy with this approach as discussed 👍